internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -546,13 +547,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 	cm.SetConditions(xpv1.ReconcileSuccess())
 
-	if !resource.IsConditionTrue(cp.GetCondition(xpv1.TypeReady)) {
-		log.Debug("Composite resource is not yet ready")
-		record.Event(cm, event.Normal(reasonBind, "Composite resource is not yet ready"))
+	if cpReady := cp.GetCondition(xpv1.TypeReady); !resource.IsConditionTrue(cpReady) {
+		log.Debug("Composite resource is not yet ready", "reason", cpReady.Reason, "message", cpReady.Message)
```

Comment:
would it be helpful to add to the log the composite name as well?

Reply:
Could be. Not caring about the debug logs though. They are useless anyway. If we cared, we would think about info level ones.

Reply:
Note that the claim should be part of the log. The claim includes the composition reference.